markhor_tracks/src/markhor_hw_interface.cpp

```diff
@@ -8,7 +8,7 @@ MarkhorHWInterface::MarkhorHWInterface()
   , start_srv_(nh.advertiseService("start", &MarkhorHWInterface::start_callback, this))
   , stop_srv_(nh.advertiseService("stop", &MarkhorHWInterface::start_callback, this))
 {
-  private_nh.param<double>("max_speed", max_speed, 1.0);
+  private_nh.param<double>("max_speed", max_speed, 3000.0);
```

Comment:
Ici pq on utilise pas une valeur venant du launch file ?

Reply:
Juste parce que c'est ce qu'il y avait déjà ce serrait vraiment nice to have en plus des gains de controlleur
mais en fait je suis pas tant sur de la pertinence du max speed logic dans le controleur, le diff_drive_controller semble déjà s'en occuper si on le paramètre bien

Reply:
Ok donc est-ce qu'on le garde ou non ?

Reply:
Sa marche bien avec pour le moment, je le laisserais jusqu'a ce qu'on retravailles sur le tout, faudrait checker si on gardes diff_drive_controller ou si on veut faire le notre en asservissant avec le gyro, ce serrait probablement nécessaire pour que l'autonome fonctionne.
So ouais on le laisse

Reply:
Okay, d'abord est-ce que tu peux le mettre dans un variable qui est plus significative svp ?

Reply:
> faudrait checker si on gardes diff_drive_controller ou si on veut faire le notre en asservissant avec le gyro, ce serrait probablement nécessaire pour que l'autonome fonctionne.

Je ne suis pas certain de comprendre cette phrase-là. Qu'est-ce qui est le *notre* dans ton contexte et comment est-ce que ça serait nécessaire pour la fonction de contrôle automne du robot ?

Reply:
Le diff drive controller de ce que j'en comprends fait les maths assumant 0 wheel slip pour faire tourner le robot au nombre de rad/s sur l'axe du yaw, or, markhor pour tourner on a bcp de wheel slip pis des comportements bizares a cause du grip des tracks, faudrait vérifier comment husky fonctionne mais leur roues semblent bcp plus slip que nos tracks.

Quand je parles de notre propre controlleur, je parles de créer une node qui prends le cmd_vel et s'occupe de driver les différentes tracks en fonction de ça, ce serrait nécessaire a mon avis si on veut avoir une logique plus complexe pour les virages(qui sont problematiques)

Reply:
Si tu ne l'avais pas déjà lu, ça serait pertinent je crois : http://wiki.ros.org/diff_drive_controller

Je ne crois pas que le contrôleur est fait pour prendre en compte le wheel slip, cependant, ça n'a pas été un problème directement relier au contrôleur avant. Si tu peux ajouter une vidéo de markhor ou d'un autre robot qui démontre à quoi tu fais référence, ça serait apprécié. 

> Quand je parles de notre propre controlleur, je parles de créer une node qui prends le cmd_vel et s'occupe de driver les différentes tracks en fonction de ça, ce serrait nécessaire a mon avis si on veut avoir une logique plus complexe pour les virages(qui sont problematiques)

Humm, j'attendrais d'avoir plus d'information avant de faire cela. J'ai l'impression que ce n'est pas tant un problème qu'on devrait essayer de régler niveau LOG en premier. J'en parlerais avec les gars de MEC si possible et même Christian pour voir se qu'il en pense.

Reply:
ouais, faudrait en parler a christian, je suis vraiment pas rendu a faire ça, je parlais vraiment juste d'une possibilité, exemple on veux set les drive a 0% output aka coast au lieu de les driver en vitesse, sa leur permeterait de se laisser glisser peut-être plus facilement, mais encore là c'est juste une idée

Reply:
> exemple on veux set les drive a 0% output aka coast au lieu de les driver en vitesse

Ehhh j'ai l'impression qu'on mélange des concepts ici, tu parles de mettre les drives à zéro pourcent de "puissance" et ceci est *also known as* coast ? Coast, c'est qu'ils ne vont pas essayer de tenir leur possible lorsqu'il n'a pas de commande envoyée. Et ensuite tu dis «au lieu de les driver en vitesse», corrige-moi, mais le fait que ça soit Coast ou brake, c'est indépendant de comment est-ce qu'on les contrôle, non ? 

Prend le temps de t'expliqué, il y a pas de stress.

Reply:
Coast et brake font référence à la connexion électrique qui est faite aux bornes du moteur lorsqu'on envoie aucune commande, par exemple lorsqu'on est dans le allowable error ou ton controlleur envoie pas de commande, l'implementation change d'une drive a l'autre mais généralement c'est ça, coast c'est laisser le moteur en circuit ouvert, ça fait en sorte que tu as juste la friction des bearing dans le moteur et dans le système qui oppose le mouvement angulaire du moteur, le brake mode, c'est mettre les 2 bornes en circuit fermé, cela fait en sorte de transformer la force éléctromotrice générée par la vitesse de rotation du moteur en courant, ce courant la induce un torque qui s'oppose a la dirrection de rotation du moteur, donc plus le moteur vas vite, plus la force de resistance est grande, en pratique sa crée un genre de brake, mais à basses vitesses tu peux souvent backdrive le moteur, dans notre système le degearing est telement grand que je penses pas qu'on puisse le backdrive manuellement, c'est a tester.

Mais ici, brake et coast ne sont pas vraiment appliqués étant donné qu'on asservis en vitesse et donc le allowable error doit être 0 pour éviter des oscillations

LE controleur en vitesse est essentiellement le meilleur brake que tu peux avoir étant donné qu'il vas appliquer un courrant au moteur quand il vas sense qu'une force externe est appliquée.

dans notre cas. je ne suis pas sur que c'est le meilleur comportement étant donné qu'on veux que les tracks slippent, sa reste a voir / tester ce qui est le mieux pour tourner sur 4 points d'appui

Reply:
Et pourquoi est-ce que tu pense cela ? 
> je ne suis pas sur que c'est le meilleur comportement étant donné qu'on veux que les tracks slippent

Reply:
tu veux que les tracks slippent/roulent par elles même pour permettre de virer étant donné qu'on a 4 points d'appui, le controlleur en vitesse essaye de se battre contre ça dans certains cas, c'est comme ça qu'un moteur surchauffe